File: C++/0011_container_with_most_water.cpp
class Solution{
public:
	int maxArea(vector<int>& height){
		int L = 0, R = height.size() - 1, res = 0;
		if(R <= L) return 0;
		while(L < R){
			int temp = (R - L) * min(height[R], height[L]);
			if(temp > res) res = temp;
			if(height[R] > height[L]){ ++L; }
			else{ --R; }	
		}
		return res;
	}
};

File: Python/0011_container_with_most_water.py
class Solution(object):
	def maxArea(self, height):
		L, R, res = 0, len(height) - 1, 0
		if R <= L: return 0
		while L < R:
			temp = (R - L) * min(height[R], height[L])
			if temp > res: res = temp
			if height[R] > height[L]: L += 1
			else: R -= 1
		return res
